tests/std/tests/LWG2381_num_get_floating_point/test.cpp

```diff
@@ -0,0 +1,509 @@
+        const char sep_str[]  = "0x17,222,9.0";
+        const size_t len      = sizeof(sep_str) - 1;
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(sep_str, sep_str + len + 1, instr, err, v);
+        assert(iter == sep_str + len);
+        assert(err == instr.goodbit);
+        assert(v == 0x172229.0p0);
+    }
+}
+
+// Also test non-ending unlimited grouping for integers
+template <class Integer>
+void test_int_grouping() {
+    const my_facet f(1);
+    ios instr(nullptr);
+    instr.imbue(locale(locale(), new mid_zero_numpunct));
+
+    Integer v = 0;
+    {
+        v                     = static_cast<Integer>(-1);
+        const char sep_str[]  = "17,2,9";
+        const size_t len      = sizeof(sep_str) - 1;
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(sep_str, sep_str + len + 1, instr, err, v);
+        assert(iter == sep_str + len);
+        assert(err == instr.goodbit);
+        assert(v == static_cast<Integer>(1729));
+    }
+    {
+        v                     = static_cast<Integer>(-1);
+        const char sep_str[]  = "17,222,9";
+        const size_t len      = sizeof(sep_str) - 1;
+        ios_base::iostate err = instr.goodbit;
+        const char* iter      = f.get(sep_str, sep_str + len + 1, instr, err, v);
+        assert(iter == sep_str + len);
+        assert(err == instr.goodbit);
+        assert(v == static_cast<Integer>(172229));
+    }
+    {
+        v                     = static_cast<Integer>(-1);
+        const char sep_str[]  = "0x17,2,9";
+        const size_t len      = sizeof(sep_str) - 1;
+        ios_base::iostate err = instr.goodbit;
+
+        instr.flags(std::ios_base::fmtflags{});
+        const char* iter = f.get(sep_str, sep_str + len + 1, instr, err, v);
+        assert(iter == sep_str + len);
+        assert(err == instr.goodbit);
+        assert(v == static_cast<Integer>(0x1729));
+    }
+    {
+        v                     = static_cast<Integer>(-1);
+        const char sep_str[]  = "0x17,222,9";
+        const size_t len      = sizeof(sep_str) - 1;
+        ios_base::iostate err = instr.goodbit;
+
+        instr.flags(std::ios_base::fmtflags{});
+        const char* iter = f.get(sep_str, sep_str + len + 1, instr, err, v);
+        assert(iter == sep_str + len);
+        assert(err == instr.goodbit);
+        assert(v == static_cast<Integer>(0x172229));
+    }
+}
+
+// Also test GH-3375 <xlocnum>: Incorrect rounding when parsing long hexadecimal floating point numbers just above
+// midpoints
+// And GH-3376 <xlocnum>: Incorrect result when parsing 9.999999...
+template <class Flt>
+void test_gh3375_gh3376() {
+    // Ensure long hexadecimal FP representations just above midpoints are correctly parsed.
+    if (is_same_v<Flt, float>) {
+        {
+            istringstream stream("0x1.000001" + string(800, '0') + "1p+0");
+            Flt x = 0.0;
+
+            stream >> x;
+            assert(bool(stream));

```

Comment:
This is a functional-style C-semantics cast, which we conventionally avoid, even in test code. Should be `static_cast<bool>`, occurs below.